pkg/controllers/tagging/tagging_controller.go

```diff
@@ -141,9 +147,15 @@ func (tc *Controller) process() bool {
 
 		err := workItem.action(workItem.node)
 		if err != nil {
-			// Put the item back on the workqueue to handle any transient errors.
-			tc.workqueue.AddRateLimited(workItem)
-			return fmt.Errorf("error finishing work item '%v': %s, requeuing", workItem, err.Error())
+			if workItem.requeuingCount < maxRequeuingCount {
```

Comment:
Can we add unit tests which verify this behavior?

Reply:
done